datalad/cmd.py

```diff
@@ -349,6 +289,10 @@ def run(self, cmd, protocol=None, stdin=None, cwd=None, env=None, **kwargs):
         FileNotFoundError
           When a given executable does not exist.
         """
+
+        # imported here to avoid recursive importing
```

Comment:
note: is happening likely due to `from .cmd import WitlessProtocol` there used solely for typing annotation.  Could have been resolved by moving protocols into their own `cmd_protocols.py` breaking the loop and making it all a bit more kosher.  I will leave to @christian-monch to decide on his cuisine .

Reply:
We do it the proper way, as you suggested: extract the protocols into their own file.